### ISSUE TITLE
1.10.0b1 test error on Mac OS X Version 10.9.5

### ISSUE THREAD

**Issue by WarrenWeckesser** [numpy/numpy#6304]
I get one error using 1.10.0b1 on a Mac with OSX 10.9.5:

```
$ python -c "import numpy as np; np.test('full', verbose=3)"
Running unit tests for numpy
NumPy version 1.10.0b1
NumPy relaxed strides checking option: True
NumPy is installed in /Users/warren/local_scipy/lib/python2.7/site-packages/numpy
Python version 2.7.10 |Anaconda 1.8.0 (x86_64)| (default, May 28 2015, 17:04:42) [GCC 4.2.1 (Apple Inc. build 5577)]
nose version 1.3.6
nose.config: INFO: Ignoring files matching ['^\\.', '^_', '^setup\\.py$']
nose.config: INFO: Excluding tests matching ['f2py_ext', 'f2py_f90_ext', 'gen_ext', 'pyrex_ext', 'swig_ext']
...<snip>...
======================================================================
ERROR: test_accelerate_framework_sgemv_fix (test_multiarray.TestDot)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/warren/local_scipy/lib/python2.7/site-packages/numpy/core/tests/test_multiarray.py", line 4218, in test_accelerate_framework_sgemv_fix
    m = aligned_array(100, 15, np.float32)
  File "/Users/warren/local_scipy/lib/python2.7/site-packages/numpy/core/tests/test_multiarray.py", line 4200, in aligned_array
    d = np.dtype()
TypeError: Required argument 'dtype' (pos 1) not found

----------------------------------------------------------------------
Ran 6090 tests in 97.453s

FAILED (KNOWNFAIL=3, SKIP=9, errors=1)
```

**Comment by charris** [comment]
Known problem fixed in 1.10.x. You can fix it by replacing `d = np.dtype()` by `d = np.dtype(0)`.

**Comment by WarrenWeckesser** [comment]
`1.10.0.rc1` passes all tests.
